src/main/java/org/apache/solr/benchmarks/task/SegmentMonitoringTask.java

```diff
@@ -0,0 +1,153 @@
+package org.apache.solr.benchmarks.task;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.prometheus.client.Gauge;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.BasicResponseHandler;
+import org.apache.solr.benchmarks.BenchmarkContext;
+import org.apache.solr.benchmarks.BenchmarksMain;
+import org.apache.solr.benchmarks.ControlledExecutor;
+import org.apache.solr.benchmarks.beans.TaskByClass;
+import org.apache.solr.benchmarks.prometheus.PrometheusExportManager;
+import org.apache.solr.benchmarks.solrcloud.SolrCloud;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.*;
+
+/**
+ * A task to report segment info of a given collection via prometheus listener/exporter
+ */
+public class SegmentMonitoringTask extends AbstractTask<List<SegmentMonitoringTask.SegmentInfo>> {
```

Comment:
Is there an example of how you would use this task? Could we add one to `suites`?

Reply:
TBH, this implementation is quite specific to our merge policy investigation so I don't want to put too much descriptions about it so we can keep README.md reasonably compact :)

The usage is in the example [here](https://github.com/fullstorydev/solr-bench/pull/88/files#diff-e67fec99eccddf4d9d8aa1cda7cffe82d5b10c118d19be1da11b9567e520603bR9-R24)